tests/program_analysis/test_py_ast_to_cast.py

```diff
@@ -0,0 +1,846 @@
ource_refs": [{"col_end": 21, "col_start": 0, "node_type": "SourceRef", "row_end": 8, "row_start": 1, "source_file_name": "test_class_2.py"}]}, {"body": [{"left": {"node_type": "Var", "source_refs": [{"col_end": 5, "col_start": 4, "node_type": "SourceRef", "row_end": 11, "row_start": 11, "source_file_name": "test_class_2.py"}], "type": "integer", "val": {"name": "f", "node_type": "Name", "source_refs": [{"col_end": 5, "col_start": 4, "node_type": "SourceRef", "row_end": 11, "row_start": 11, "source_file_name": "test_class_2.py"}]}}, "node_type": "Assignment", "right": {"arguments": [], "func": {"name": "Foo", "node_type": "Name", "source_refs": [{"col_end": 13, "col_start": 8, "node_type": "SourceRef", "row_end": 11, "row_start": 11, "source_file_name": "test_class_2.py"}]}, "node_type": "Call", "source_refs": [{"col_end": 13, "col_start": 8, "node_type": "SourceRef", "row_end": 11, "row_start": 11, "source_file_name": "test_class_2.py"}]}, "source_refs": [{"col_end": 13, "col_start": 4, "node_type": "SourceRef", "row_end": 11, "row_start": 11, "source_file_name": "test_class_2.py"}]}, {"expr": {"arguments": [], "func": {"attr": {"name": "bar", "node_type": "Name", "source_refs": [{"col_end": 9, "col_start": 4, "node_type": "SourceRef", "row_end": 12, "row_start": 12, "source_file_name": "test_class_2.py"}]}, "node_type": "Attribute", "source_refs": [{"col_end": 9, "col_start": 4, "node_type": "SourceRef", "row_end": 12, "row_start": 12, "source_file_name": "test_class_2.py"}], "value": {"name": "f", "node_type": "Name", "source_refs": [{"col_end": 5, "col_start": 4, "node_type": "SourceRef", "row_end": 12, "row_start": 12, "source_file_name": "test_class_2.py"}]}}, "node_type": "Call", "source_refs": [{"col_end": 11, "col_start": 4, "node_type": "SourceRef", "row_end": 12, "row_start": 12, "source_file_name": "test_class_2.py"}]}, "node_type": "Expr", "source_refs": [{"col_end": 11, "col_start": 4, "node_type": "SourceRef", "row_end": 12, "row_start": 12, "source_file_name": "test_class_2.py"}]}], "func_args": [], "name": "main", "node_type": "FunctionDef", "source_refs": [{"col_end": 11, "col_start": 0, "node_type": "SourceRef", "row_end": 12, "row_start": 10, "source_file_name": "test_class_2.py"}]}, {"expr": {"arguments": [], "func": {"name": "main", "node_type": "Name", "source_refs": [{"col_end": 6, "col_start": 0, "node_type": "SourceRef", "row_end": 14, "row_start": 14, "source_file_name": "test_class_2.py"}]}, "node_type": "Call", "source_refs": [{"col_end": 6, "col_start": 0, "node_type": "SourceRef", "row_end": 14, "row_start": 14, "source_file_name": "test_class_2.py"}]}, "node_type": "Expr", "source_refs": [{"col_end": 6, "col_start": 0, "node_type": "SourceRef", "row_end": 14, "row_start": 14, "source_file_name": "test_class_2.py"}]}], "name": "test_class_2", "node_type": "Module", "source_refs": [{"col_end": null, "col_start": null, "node_type": "SourceRef", "row_end": 14, "row_start": 1, "source_file_name": "test_class_2.py"}]}]}'
```

Comment:
@titomeister same comment as above. If these are expected result JSONs then let's save them in a dir labeled `expected_output/` or something like that.